### PR TITLE
feat: [pipe-22241]: add compare btn on branch list- on gitness

### PR DESCRIPTION
**Pull request by cjlee01**
add gitness integration for the button
- handle routing with or without diffRefs
- add messaging if head commit is the same
change formfield name
https://github.com/user-attachments/assets/a9635289-eac6-464e-af06-3edaf7cbe477

### CODE REVIEW

File: apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx [hunk 1]
```diff
@@ -27,20 +27,27 @@ import * as Diff2Html from 'diff2html'
 
 export const CreatePullRequest = () => {
   const createPullRequestMutation = useCreatePullReqMutation({})
-  const { repoId, spaceId } = useParams<PathParams>()
+  const { repoId, spaceId, diffRefs } = useParams<PathParams>()
+  const [isBranchSelected, setIsBranchSelected] = useState<boolean>(diffRefs ? true : false) // State to track branch selection
+
+  // console.log(diffRefs)
+  const diffTargetBranch = diffRefs?.split('...')[0] || ''
+  const diffSourceBranch = diffRefs?.split('...')[1] || ''
```
Comment (vardanbansal-harness) [inline apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx:35]: Probably `const [diffTargetBranch, diffSourceBranch] = diffRefs?.split('...')` ?

File: apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx [hunk 2]
```diff
@@ -185,9 +192,14 @@ export const CreatePullRequest = () => {
...
     // useMergeCheckMutation
-    mergeCheck({ queryParams: {}, repo_ref: repoRef, range: diffApiPath }).then(value => {
-      setMergeabilty(value?.mergeable)
-    })
+    mergeCheck({ queryParams: {}, repo_ref: repoRef, range: diffApiPath })
+      .then(value => {
+        setMergeabilty(value?.mergeable)
+      })
+      .catch(() => {
+        console.log('eerr')
```
Comment (vardanbansal-harness) [inline apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx:200]: remove this?

**Review by vardanbansal-harness — APPROVED** [review]

File: apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx (hunk 2 above)
Comment (cjlee01) [reply apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx:200]: fixed

File: apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx (hunk 1 above)
Comment (cjlee01) [reply apps/gitness/src/pages/pull-request/pull-request-compare-page.tsx:35]: fixed

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**